web/pgadmin/tools/sqleditor/__init__.py

```diff
@@ -950,6 +950,14 @@ def poll(trans_id):
                 is_thread_alive = True
                 break
 
+    # if transaction object is instance of QueryToolCommand
+    # and transaction aborted for some reason then issue a
+    # rollback to cleanup
+    if isinstance(trans_obj, QueryToolCommand):
```

Comment:
For any particular scenario, did you put this check before polling?

Reply:
Yes there was one scenario, where rollback is happening after we are getting the transaction_in_error status. so the response holds the old status. so I in place of putting this check in every if or elif block, I put it before polling.